python/nano/src/bigdl/nano/utils/common/optimizer/optimizer.py

```diff
@@ -160,5 +172,49 @@ def get_best_model(self,
             invalidInputError(False,
                               "Don't find related model in optimize's results.")
 
+        if search_env:
+            self._search_env(method_name=best_metric.method_name)
+
         return best_model, format_acceleration_option(best_metric.method_name,
                                                       self.ALL_INFERENCE_ACCELERATION_METHOD)
+
+    def _search_env(self, method_name: str = None):
+        invalidInputError(method_name,
+                          'Must give specific model method for searching env.')
+        invalidInputError(method_name in self.optimized_model_dict,
+                          'Must provide correct method name.')
+        if 'env' not in self.optimized_model_dict[method_name]:
+            print(f"=================Start searching for environment variables "
+                  f"for {method_name} model=================")
+            env_result_map = {}
+            for idx, (env_name, env) in enumerate(self.ALL_ACCELERATION_ENV.items()):
+                print(f"----------Start test {env_name} variables "
+                      f"({idx + 1}/{len(self.ALL_ACCELERATION_ENV)})----------")
+                try:
+                    env_result_map[env_name] = self._latency_calc_with_worker(
+                        self.optimized_model_dict[method_name]['model'], env=env.get_env_dict())
+                except Exception as e:
+                    print("----------worker failed to execute----------")
+                    print(e.args)
+                    traceback.print_exc()
+                    print(f"----------Failed to run with {env_name} variables----------")
+            for env_name, latency in env_result_map.items():
+                print(f"{env_name}\t{latency} ms")
+            best_env = self.ALL_ACCELERATION_ENV[
+                min(env_result_map, key=env_result_map.get)].get_env_dict()
+            self.optimized_model_dict[method_name]['env'] = best_env
+        env_suggestion = '\n'.join(f'export {env_key}=\'{env_value}\''
+                                   for env_key, env_value in
+                                   self.optimized_model_dict[method_name]['env'].items())
+        register_suggestion(f'You can try the following commands for better performance\n'
+                            f'{env_suggestion}')
```

Comment:
the suggestion could be dump to a sh file as we talked about, but we may do it in next PR.